java/readme.md

```diff
@@ -0,0 +1,213 @@
+# Java code example
+
+
+## Prepare the client environment
+
+Download and install the JDK from [openjdk.org](https://openjdk.org/projects/jdk/21/) and Maven from [maven.apache.org](https://maven.apache.org/download.cgi). This example uses the [Apache Kafka](https://github.com/apache/kafka) client library.
+
+```bash
+# Create and enter the project folder
+mkdir redpanda-java; cd redpanda-java
+# Initialize the project
+mkdir -p src/main/java/org/example
+cat > pom.xml <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.example</groupId>
+    <artifactId>redpanda-client</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```

Comment:
Nit: Why does the code sample require Java 21? Does it mean that it doesn't run on older Java?

Reply:
21 is the most recent LTS version, but I know lots of places may be "stuck" with older stuff. I'll add a comment to that effect.